File: kiosconeta-backend/Domain/Entities/Venta.cs
namespace Domain.Entities
{
    public class Venta
    {
        public int VentaId { get; set; }

        public DateTime Fecha { get; set; }

        public decimal PrecioCosto { get; set; }

        /// <summary>Suma de precios de venta de todos los productos (sin descuento)</summary>
        public decimal Subtotal { get; set; }

        /// <summary>Descuento aplicado (combos, promos, manual)</summary>
        public decimal Descuento { get; set; } = 0;

        /// <summary>Total final = Subtotal - Descuento</summary>
        public decimal Total { get; set; }

        public string? Detalles { get; set; }

        public int EmpleadoId { get; set; }
        public Empleado Empleado { get; set; }

        public int CierreTurnoId { get; set; }
        public CierreTurno CierreTurno { get; set; }

        public int MetodoPagoId { get; set; }
        public MetodoDePago MetodoPago { get; set; }

        public int TurnoId { get; set; }
        public Turno Turno { get; set; }

        public int NumeroVenta { get; set; }
        public bool Anulada { get; set; }
        public IList<ProductoVenta> ProductoVentas { get; set; }
    }
}

File: kiosconeta-frontend/src/types/venta.ts
// ════════════════════════════════════════════════════════════════════════════
// TYPES: Ventas
// ════════════════════════════════════════════════════════════════════════════

// ────────────────────────────────────────────────────────────────────────────
// VENTA (Entity completa)
// ────────────────────────────────────────────────────────────────────────────

export interface Venta {
  ventaId: number;
  fecha: string;
  total: number;
  precioCosto: number;
  ganancia: number;
  margenGanancia: number;
  detalles: string | null;
  anulada: boolean;
  numeroVenta: number;
  // Empleado
  empleadoId: number;
  empleadoNombre: string;
  // Método de pago
  metodoPagoId: number;
  metodoPagoNombre: string;
  // Turno
  turnoId: number;
  turnoNombre: string;
  // Productos
  productos: ProductoVenta[];
}

export interface ProductoVenta {
  productoVentaId: number;
  productoId: number;
  productoNombre: string;
  cantidad: number;
  precioUnitario: number;
  subtotal: number;
}

// ────────────────────────────────────────────────────────────────────────────
// CREAR VENTA (DTO para el POS)
// ────────────────────────────────────────────────────────────────────────────

export interface CreateVentaDTO {
  empleadoId: number;
  metodoPagoId: number;
  turnoId: number;
  detalles?: string;        // Observaciones opcionales
  productos: CreateProductoVentaDTO[];
}

export interface CreateProductoVentaDTO {
  productoId: number;
  cantidad: number;
  // precioUnitario lo toma el backend del producto directamente
}

// ────────────────────────────────────────────────────────────────────────────
// CARRITO (para el POS)
// ────────────────────────────────────────────────────────────────────────────

export interface ItemCarrito {
  productoId: number;
  nombre: string;
  precioUnitario: number;
  cantidad: number;
  subtotal: number;
  stock: number;              // Stock disponible
}

export interface Carrito {
  items: ItemCarrito[];
  subtotal: number;
  descuento: number;
  total: number;
  metodoPagoId?: number;
}

// ────────────────────────────────────────────────────────────────────────────
// MÉTODO DE PAGO
// ────────────────────────────────────────────────────────────────────────────

export interface MetodoPago {
  metodoDePagoID: number;   // el backend devuelve "metodoDePagoID" con mayúscula
  nombre: string;
}

// ────────────────────────────────────────────────────────────────────────────
// FILTROS DE VENTAS
// ────────────────────────────────────────────────────────────────────────────

export interface VentaFiltros {
  fechaDesde?: string;        // ISO date string
  fechaHasta?: string;        // ISO date string
  empleadoId?: number;
  metodoPagoId?: number;
  incluirAnuladas?: boolean;
}